app/classes/Transvision/Utils.php

```diff
@@ -438,10 +438,32 @@ public static function ago($datetime, $ref_time = '')
      * This is used on views which also exist in our public API
      * https://github.com/mozfr/transvision/wiki/JSON-API
      *
-     * @return string URL with 'json' appended as part of the query string
+     * @param  boolean $revert if true, invert the locale and the source_locale in the API generated link
+     * @return string  URL with 'json' appended as part of the query string
      */
-    public static function redirectToAPI()
+    public static function redirectToAPI($revert = false)
     {
-        return $_SERVER["REQUEST_URI"] . (is_null($_SERVER['QUERY_STRING']) ? '?json' : '&json');
+        if (! is_null($_SERVER['QUERY_STRING'])) {
+            if ($revert == true) {
+                $arg = [];
+                parse_str($_SERVER['QUERY_STRING'], $arg);
+                $source_locale = $arg["sourcelocale"];
+                $locale = $arg["locale"];
+                $arg["sourcelocale"] = $locale;
+                $arg["locale"] = $source_locale;
+                $query = http_build_query($arg);
```

Comment:
I'd wait for @pascalchevrel to see his opinion on the general approach. Having said that, this is clearer but we can swap the values with just one temp variable (also add a comment?).